### ISSUE TITLE
Offline version?

### ISSUE THREAD

**Issue by masternetra** [silexlabs/Silex#568]
I was wondering if there is or will be an offline version of this editor, not all of us can be connected to the net all the time?

**Comment by singchan** [comment]
+1
I understood that @JbIPS and @detectedstealth had the intention of giving it a try

**Comment by JbIPS** [comment]
Yes, I pushed some work in progress on the [GitLab](https://framagit.org/JbIPS/Silex/tree/electron) to make Silex into an Electron app. For now, I didn't handle the file operations (save, load,...) but it's only a start 😄 .

Any help is welcome!

**Comment by detectedstealth** [comment]
Yeah I am about to start working on something using electron as well so @JbIPS I'll take a look at what you are doing.

**Comment by JbIPS** [comment]
Thank you, that's my first electron app so any feedback will be very appreciated!

**Comment by lexoyo** [comment]
We need testers for this
@JbIPS has made a great first version

**Comment by JbIPS** [comment]
If you're on Linux (Debian or Red Hat based), you could try the first release of Silex here : https://github.com/silexlabs/Silex/releases/tag/v1.0.0-rc1. Your feedback is very welcome!

**Comment by AndreaMonzini** [comment]
Hello, i'm testing Silex on Debian based Linux distribution as Electron App and it seems to work.
Can i have access with the desktop app to the local files and assets instead of using FTP? it would be very useful.
Thank you.

**Comment by JbIPS** [comment]
Hi @AndreaMonzini! Thank you for your feedback, it's a motivation to keep going on! 

I'm surprise you cannot use local files since it was designed especially for this... What are your choices when opening a file? If you have nothing like "local" or "fs" it may be a bug and I'll look into it.

**Comment by AndreaMonzini** [comment]
I cannot find where to load local files:

![screenshot from 2017-11-04 02-18-59](https://user-images.githubusercontent.com/12413409/32401150-be9e8ee2-c109-11e7-82e5-aeb3341c4731.png)

![screenshot from 2017-11-04 02-21-14](https://user-images.githubusercontent.com/12413409/32401154-c6a7ab28-c109-11e7-8d1c-af246f798e8b.png)

**Comment by JbIPS** [comment]
I see, there is a mistake in the labels of this version. "Web server" is actually your local disk. We renamed that on the newest version but I didn't published them.

You can use it as-is until we publish the new version (when preprod will go in prod).

**Comment by AndreaMonzini** [comment]
I tried with my user name ( root)  and password but i cannot access:

![screenshot from 2017-11-04 12-43-42](https://user-images.githubusercontent.com/12413409/32405163-e2b1e45e-c15f-11e7-945d-f05facada2a4.png)

**Comment by JbIPS** [comment]
I'm sorry, it's actually a conf in Silex. Use admin/admin

**Comment by AndreaMonzini** [comment]
it works! Thank you !
I will test a Website based on a template!
What version of Silex is used with Electron? If possible, i can test last version.

![screenshot from 2017-11-04 14-03-59](https://user-images.githubusercontent.com/12413409/32405642-b2654fde-c169-11e7-8f83-5d058e56ec69.png)

**Comment by JbIPS** [comment]
@lexoyo will confirm that but I think that's the actual online version. We'll try to keep it in sync with the live version.

Thank you again for beta testing,  we'll warn new when new versions come out.

**Comment by lexoyo** [comment]
> @lexoyo will confirm that but I think that's the actual online version. We'll try to keep it in sync with the live version.

It looks like it. But I don't know how you build it? Is it with `npm run electron`?
Will it be automatically built and deployed when we merge into master?

**Comment by lexoyo** [comment]
I tried and `npm start:electron` does launch the electron version
Should we publish Silex on npm and let people do `npm install -g silexv2 && npm run start:electron` ?

**Comment by JbIPS** [comment]
What I add in mind was to add a CI instruction in Silex that would build and publish a new release of the  desktop app every time you merge into master.

But yes, for the beta testers, this command (without the `-g` though) will be great.

**Comment by AndreaMonzini** [comment]
Hello, when i try to publish to the local folder "Pub" ( selected in the publishing settings), the images are saved corrupted and not visible in the published website.
After i overwrite manually the images, they became visible.

![screenshot from 2017-11-04 16-19-16](https://user-images.githubusercontent.com/12413409/32406675-ed6a22c2-c17c-11e7-941d-2bd8458fe409.png)

**Comment by lexoyo** [comment]
This looks like an old bug, you should test the new version... Give us some time to build it or clone the repo, follow the installation instructions and run the command
$ npm run start:electron

**Comment by AndreaMonzini** [comment]
I wait the new version, thank you!

**Comment by AndreaMonzini** [comment]
Hello, i'm testing silex electron for my website:

http://monzini.art/

The desktop version seems ok with some glitches
But the mobile layout is wrong: very different from the mobile version i see in the edit.

Also with mobile i see problem with redirect, i need to check the server configurations.

I noticed also some problem adding images as i see an offset on the right, while if i use the container it works.( example in the image attached )


![screenshot from 2017-11-08 11-30-32](https://user-images.githubusercontent.com/12413409/32545130-0030bf96-c47b-11e7-8c75-eb554a21e52e.png)

**Comment by lexoyo** [comment]
Hi @AndreaMonzini 

> The desktop version seems ok with some glitches !

Please open separate issues for each "glitch" 

> But the mobile layout with iOS® ( Safari®, Firefox®, Chrome®) is wrong: very different from the mobile version that i see in the mobile edit, not usable with all browsers.
While with Android ® the layout is similar to the mobile edit ( some problems but usable)

it looks like the css are not loaded at all... is it a published website?

> I noticed also some problem adding images as i see an offset on the right, while if i use the container it works.( example in the image attached , the first logo is container added , the second logo image added with visible offset )

i can not reproduce, would you share the website with me please?
also you can try the https://preprod.silex.me version, it has bug fixes (only the publication does not work yet)

> Anyway thank you for Silex on Electron for Linux and Galaxee template!

👍 

> Are the attributions visible and correct at the end of the website?

I don't see them but you do not need to credit us anyway :)

**Comment by AndreaMonzini** [comment]
Hello, 
i had to redirect the website.
Silex website link:

http://monzini.art/silex/

**Comment by lexoyo** [comment]
>  in Electron version i was not able to disable the mobile version ( checkbox was not visible) 

oh wow this is strange
can you open an issue please?

> with https://preprod.silex.me i was not able to publish

to publish on ftp or dropbox?

> A solution could be an updated Electron version :)

As soon as preprod is released I think @JbIPS will take care of this
